projects/components/src/select/select.component.ts

```diff
@@ -90,13 +100,16 @@ export class SelectComponent<V> implements AfterContentInit, OnChanges {
   @Input()
   public highlightSelected: boolean = true;
 
+  @Input()
+  public multiSelectMode: boolean = false;
```

Comment:
All of the inputs/outputs need to support arrays now, right? or unlink them, support both in, and just emit the selection change each time (but would require metadata saying if it was selected or unselected)

Reply:
This is the reason why I was working with an OR'ed type instead of taking in arrays. Any usages of the emitted events in all consumers have to be modified. 
In that case, do you think a separate component would be better? @aaron-steinfeld

Reply:
Given how similar they are, I'd lean away from a separate components.
So we could:
1. update all the consumers (hopefully not too many).
2. Have a union of types on inputs + outputs (in this model, I'd at least convert to arrays internally to keep it clean there. Could also make the union a generic type, so the caller can narrow it based on their use case.
3. Have different outputs (and potentially inputs) for different modes. (this strikes me as bad)

I'd probably lean towards 2 (iterating on what we have here)
What are other thoughts @anandtiwary @jake-bassett ?

Reply:
Number 2 was what I was aiming for. But after making these changes and seeing whats ahead, I think a different component fits better.
In case of `selectedChange$` for single select, the change always has one value. In case of multi-select, this contract changes. The consumers will have to take the burden of casting appropriately and consuming the values.
Similarly, some styling changes also does not apply for multi select. 

More I think of this, I feel like its cleaner to have a separate multi select component. I don't think we gain much by trying to reuse code with union types. Code (both template and TS) will be messier with more casting / if else blocks.

Reply:
Sure, let's go with the separate components then.

Reply:
+1 to separate components